### ISSUE TITLE
is it possible to change port without restarting?

### ISSUE THREAD

**Issue by piramiday** [rakshasa/rtorrent#1015]
at startup, my rtorrent.rc file specifies a non-random listening port, e.g. 54321.
rtorrent will start and will announce itself to the tracker, with ip 1.2.3.4 and `client_id` similar to `-lt0D80-WHATEVER`.
rtorrent will then start seeding and keep announcing uploaded data to the tracker.

let's then assume **a disconnection happens while rtorrent is still alive**: whenever connectivity is resumed, the pc with rtorrent will have a different ip and port.
**is there a way to let rtorrent acknowledge this without restarting?**

in other words, I would like to **keep being** `client_id=-lt0D80-WHATEVER` for the tracker, but I would like to start announcing a different ip address and port, say, 5.6.7.8 with port 12345.

is it possible, maybe through some clever combination of CTRL+X commands?

**Comment by FraMecca** [comment]
Yes it is possible:
https://github.com/rakshasa/rtorrent/issues/700
https://github.com/rakshasa/rtorrent/issues/211

Another possibility however is doing port fowarding through a router / ssh

**Comment by piramiday** [comment]
on the contrary, it does not look like it's possible -- the `iptables` command is a rough workaround.
the proper solution would be some CTRL-X command through which rtorrent forcefully closes the listening port and reopens it on a different port.
